### PR TITLE
BL-729 Updated Matcher Algorithm

### PR DESCRIPTION
**Pull request by stevenhastings**
## Description
Updated MatcherSortSearch class within model.py to account for 3 relevant fields:
*   1.) "is_active" == type(bool) whether or not mentee or mentor have active accounts.
*   2.) "accepting_new_mentees" == type(bool) whether or not mentor is currently taking on new mentees. 
*   3.) "validate_status" == type(str) application status classified as one of ["approved", "rejected", "pending"]

## Jira Ticket
[BL-729 Jira Ticket](https://bloomtechlabs.atlassian.net/browse/BL-729)

## Type of Change
- [x] New feature

## Checklist
- [x] My code follows PEP8 style guide
- [x] I have removed unnecessary print statements from my code
- [x] I have made corresponding changes to the documentation if necessary
- [x] My changes generate no errors
- [x] No commented-out code
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

## Loom Video
[Steven's LoomTech Talk](https://www.loom.com/share/b99f8081e5e647b5ae1bcf67c1643545)

### CODE REVIEW

File: app/model.py [hunk 1]
```diff
@@ -19,7 +19,12 @@ def sort_mentors(mentor: Dict) -> Tuple:
             )
 
         results = sorted(
-            self.db.search("Mentors", mentee["tech_stack"]),
+            self.db.read("Mentors", {
+                "tech_stack": mentee["tech_stack"],
+                "is_active": True,
+                "accepting_new_mentees": True,
+                "validate_status": 'approved'
+                }),
```
Comment (BrokenShell) [inline app/model.py:27]: One tiny PEP8 issue here on line 27, unindent once to make it aligned with the opening brackets on line 22.

Comment (BrokenShell) [reply app/model.py:27]: Or Command-Option-L to make PyCharm do it for you.

**Review by woodbenwood — COMMENTED** [review]
.

**Review by woodbenwood — DISMISSED** [review]
Great, we've avoided the potential merge conflict from earlier today. And I believe this is verbatim the code that was agreed upon. Happy to approve -- this is going to be a fantastic match mentees feature and will really pop whence visualized. Excited.

File: app/model.py (hunk 1 above)
Comment (woodbenwood) [reply app/model.py:27]: this is actually a sick burn since Steven is still on vscode...

**Review by BrokenShell — APPROVED** [review]
This is fantastic! It will filter out any Mentor who isn't a viable choice for matching.

File: app/model.py [hunk 2]
```diff
@@ -19,7 +19,12 @@ def sort_mentors(mentor: Dict) -> Tuple:
             )
 
         results = sorted(
-            self.db.search("Mentors", mentee["tech_stack"]),
+            self.db.read("Mentors", {
+                "tech_stack": mentee["tech_stack"],
+                "is_active": True,
+                "accepting_new_mentees": True,
+                "validate_status": 'approved'
+            }),
```
Comment (woodbenwood) [inline app/model.py:27]: Great, we've avoided the potential merge conflict from yesterday. Except for a humiliating indentation error that we should all be ashamed of, this was done perfectly. Happy to approve -- this is going to be a fantastic match mentees feature and will really pop whence visualized. Excited.

**Review by woodbenwood — APPROVED** [review]

**Review by L0lbrich — APPROVED** [review]
This is great! I can't wait to see this be used on real data

**Pull request merged**